tls/s2n_ktls.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/s2n_ktls.h"
+
+/* Compares s2n_ktls_mode to see if the original_state implies the check_state.
+ *
+ * original_state           check_state                                 requested
+ * ------------------------------------------------------------------------------
+ * S2N_KTLS_MODE_DUPLEX     S2N_KTLS_MODE_DUPLEX                        true
+ * S2N_KTLS_MODE_SEND       S2N_KTLS_MODE_SEND | S2N_KTLS_MODE_DUPLEX   true
+ * S2N_KTLS_MODE_RECV       S2N_KTLS_MODE_RECV | S2N_KTLS_MODE_DUPLEX   true
+ * S2N_KTLS_MODE_DISABLED   S2N_KTLS_MODE_DISABLED                      true
+ *
+ */
+S2N_RESULT s2n_ktls_is_mode(s2n_ktls_mode original_state, s2n_ktls_mode check_state, bool *is_true)
+{
+    if (check_state == S2N_KTLS_MODE_DUPLEX) {
+        *is_true = original_state == S2N_KTLS_MODE_DUPLEX;
+    } else if (check_state == S2N_KTLS_MODE_DISABLED) {
+        *is_true = original_state == S2N_KTLS_MODE_DISABLED;
+    } else {
+        *is_true = original_state & check_state;
+    }
+
+    return S2N_RESULT_OK;
+}
```

Comment:
I'm confused by this method. Maybe I'm just missing context, since whatever internal method requires it hasn't been created yet? What's the use case for this method?

Minimally, I'd probably change the name to "s2n_ktls_mode_is_mode". But if I'm understanding what you're trying to do with this method, then I think it would be simpler to just have two bitfields on the config / connection, one for send and one for receive. The enum would only be for the application to pass to the setter.